### DISCUSSION TITLE
Struggling to get data from SeaOrm to render in Handlebars

### DISCUSSION THREAD

**Discussion by dave42w** [sunng87/handlebars-rust#559] [Q&A] [answered]
Hi,

Sorry this is very basic.

I have vector of structs (and it is serializable):

let places:Vec<place::Model> = Place::find().all(&s.db).await
    .expect("could not find any places");
    println!("places: {:?}",places );

That prints as:
places: [Model { id: 1, name: "London" }, Model { id: 2, name: "Paris" }, Model { id: 3, name: "New York" }, Model { id: 4, name: "New York" }, Model { id: 5, name: "York" }]

I add this to a HashMap for Handlebars:
    let mut data = BTreeMap::new();
    data.insert("place".to_string(), [places]);
    println!("data: {:?}", data);
    s.templates.render("place_list.html", &data).unwrap().into()

This prints:
data: {"place": [[Model { id: 1, name: "London" }, Model { id: 2, name: "Paris" }, Model { id: 3, name: "New York" }, Model { id: 4, name: "New York" }, Model { id: 5, name: "York" }]]}

My template is:
<ul class="place_list">
  {{#each place}}
    <li>a {{name}}</li>
  {{/each}}
</ul>

At runtime I get:
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: RenderError { desc: "Cannot access array/vector with string index.", template_name: Some("place_list.html"), line_no: Some(16), column_no: Some(11), cause: Some(ParseIntError { kind: InvalidDigit }), unimplemented: false }', src/controllers/place.rs:35:50
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

This is my (auto generated) entity struct:
[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq, Serialize, Deserialize)]
#[sea_orm(table_name = "place")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub name: String,
}

I assume that my vector isn't being serialized properly? Or that I need an extra conversion step? I've struggled to find any examples 

I've tried explicitly converting to json (but wondering if I need to do this for each element in the vector) eg

   let places:Vec<place::Model> = Place::find().all(&s.db).await
    .expect("could not find any places");
    println!("places: {:?}",places );

    let json = serde_json::to_string(&places).expect("json bug");
    println!("json {:?}",json );
     
    let mut data = BTreeMap::new();
    data.insert("place".to_string(), [json]);
    println!("data: {:?}", data);
    s.templates.render("place_list.html", &data).unwrap().into()

The debugging println! s show:
places: [Model { id: 1, name: "London" }, Model { id: 2, name: "Paris" }, Model { id: 3, name: "New York" }, Model { id: 4, name: "New York" }, Model { id: 5, name: "York" }]
json "[{\"id\":1,\"name\":\"London\"},{\"id\":2,\"name\":\"Paris\"},{\"id\":3,\"name\":\"New York\"},{\"id\":4,\"name\":\"New York\"},{\"id\":5,\"name\":\"York\"}]"
data: {"place": ["[{\"id\":1,\"name\":\"London\"},{\"id\":2,\"name\":\"Paris\"},{\"id\":3,\"name\":\"New York\"},{\"id\":4,\"name\":\"New York\"},{\"id\":5,\"name\":\"York\"}]"]}

My template renders as:
<ul class="place_list">
    <li>a </li>
</ul>

I'm new to rust and all these crates, I also haven't done professional coding for a long time and am trying to get back to it. So I am probably being very slow and stupid.

Thanks

Dave

**Comment by sunng87** [comment, marked as answer]
> data: {"place": [[Model { id: 1, name: "London" }, Model { id: 2, name: "Paris" }, Model { id: 3, name: "New York" }, Model { id: 4, name: "New York" }, Model { id: 5, name: "York" }]]}

You seem to wrap places in nested vectors, try to fix it like this:

```rust
let mut data = BTreeMap::new();
// use places directly
data.insert("place".to_string(), places);
println!("data: {:?}", data);
s.templates.render("place_list.html", &data).unwrap().into()
```

And you cannot serialize your data to json to render in handlebars because we only use the json type system not the format.

**Reply by dave42w** [reply]
Thanks so much! Works like a charm :-)
